ci/Dockerfile

```diff
@@ -0,0 +1,77 @@
+FROM docker:dind
+
+# set labels
+LABEL maintainer="Nightah"
+
+# set application versions
+ARG ARCH="amd64"
+ARG BUILDKITE_VERSION="3.17.0"
+ARG OVERLAY_VERSION="v1.22.1.0"
+
+# environment variables
+ENV PS1="$(whoami)@$(hostname):$(pwd)$ " \
+HOME="/buildkite" \
+TERM="xterm"
+
+# set runtime variables
+ENV BUILDKITE_AGENT_CONFIG=/buildkite/buildkite-agent.cfg \
+GOPATH="/buildkite/.go"
+
+# add local files
+COPY root/ /
+
+# modifications
+RUN \
+ echo "**** Install Authelia CI pre-requisites ****" && \
+   echo "@edge http://dl-cdn.alpinelinux.org/alpine/edge/community" >> /etc/apk/repositories && \
+   echo "@edget http://dl-cdn.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories && \
+   apk add --no-cache \
+     bash \
+     ca-certificates \
+     coreutils \
+     chromium \
+     chromium-chromedriver \
+     curl \
+     docker-compose \
+     git \
+     hub@edget \
+     go@edge \
+     jq \
+     libc6-compat \
+     libstdc++ \
+     nodejs \
+     npm \
+     openssh-client \
+     perl \
+     rsync \
+     shadow \
+     sudo \
+     tzdata \
+     yarn@edge && \
+ echo "**** Add s6 overlay ****" && \
+   cd /tmp && \
+   curl -Lfs -o s6-overlay.tar.gz "https://github.com/just-containers/s6-overlay/releases/download/${OVERLAY_VERSION}/s6-overlay-${ARCH}.tar.gz" && \
+   tar xfz s6-overlay.tar.gz -C / && \
+ echo "**** Patch CVE-2019-5021 ****" && \
+   sed -i -e 's/^root::/root:!:/' /etc/shadow && \
+ echo "**** Create abc user and make our folders ****" && \
+   useradd -u 911 -U -d /buildkite -s /bin/false abc && \
+   usermod -G wheel abc && \
+   sed -i 's/# %wheel/%wheel/g' /etc/sudoers && \
+ echo "**** Install Buildkite ****" && \
+   mkdir -p /buildkite/builds /buildkite/hooks /buildkite/plugins && \
+   curl -Lfs -o /usr/local/bin/ssh-env-config.sh https://raw.githubusercontent.com/buildkite/docker-ssh-env-config/master/ssh-env-config.sh && \
```

Comment:
please fix the version by using a tag or branch instead of master. This will prevent breaking in the future.

Reply:
This is the same as upstream's configuration for how they build their own docker container:
https://github.com/buildkite/agent/blob/f51dc916c5279d72e541c5b47b48159668ac6dea/packaging/docker/alpine-linux/Dockerfile#L24-L26

I also have notifications on when the upstream repo performs any commits to be able to track and monitor this.

Reply:
Human monitoring is not reliable. I'm ok to try like that but we'll fix the version it if breaks too often.